swim/disseminator.go

```diff
@@ -179,3 +179,24 @@ func (d *disseminator) RecordChange(change Change) {
 	d.changes[change.Address] = &pChange{change, 0}
 	d.Unlock()
 }
+
+func (d *disseminator) ClearChange(c Change) {
+	d.Lock()
+	delete(d.changes, c.Address)
+	d.Unlock()
+}
+
+func (d *disseminator) ChangeByAddress(address string) (Change, bool) {
+	d.Lock()
+	pc, ok := d.changes[address]
+	c := pc.Change
```

Comment:
What happens when the read is not `ok`? I would expect a `nil pointer dereference` here. Lets wrap this with a `if ok { ... }`